File: angular-web-task/src/app/components/academics/department/department.ts
import { Faculty } from '../faculty/faculty';

export class Department {

  id: number;
  name: string;
  code: string;
  description: string;
  faculty: Faculty;

  constructor(id?: number, name?: string, code?: string, description?: string, faculty?: Faculty) {
    this.id = id;
    this.name = name;
    this.code = code;
    this.description = description;
    this.faculty = faculty;
  }
}

File: WebTask/src/main/java/com/example/WebTask/modules/academics/department/Department.java
package com.example.WebTask.modules.academics.department;

import com.example.WebTask.modules.academics.Academics;
import com.example.WebTask.modules.academics.faculty.Faculty;
import lombok.Data;
import javax.persistence.*;


@Entity
@Data
@PrimaryKeyJoinColumn(name = "DEPARTMENT_ID")
public class Department extends Academics {



    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.MERGE)
    @JoinColumn(name = "FACULTY_ID", nullable = false)
    private Faculty faculty;
}
